CMakeLists.txt

```diff
@@ -68,7 +68,16 @@ if(CMAKE_VERSION VERSION_GREATER "3.1.0")
   endif()
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   if (NOT DEFINED CMAKE_C_STANDARD)
-    set(CMAKE_C_STANDARD 99)
+    try_compile(
+            RESULT
+            ${PROJECT_BINARY_DIR}
+            SOURCES "${CMAKE_CURRENT_LIST_DIR}/tests/compiler_features_tests/simple_main.c"
+            CMAKE_FLAGS "-DCMAKE_C_FLAGS=\"-std=c11\"")
+    if(RESULT)
+      set(CMAKE_C_STANDARD 11)
+    else()
+      set(CMAKE_C_STANDARD 99)
+    endif()
```

Comment:
NP: This block is conditioned on `CMAKE_VERSION VERSION_GREATER "3.1.0"` but I don't think that's necessary for testing compiler C11 compatibility. (We could still set the `CMAKE_C_STANDARD` and `CMAKE_CXX_STANDARD` values even if they will be ignored by CMake.)

Instead we could condition the `set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")` line below (and a C11 analog) on the `CMAKE_VERSION` being at least 3.1.